models/evc.py

```diff
@@ -1410,6 +1410,63 @@ def handle_link_down(self):
 
         return success
 
+    @staticmethod
+    def get_interface_from_switch(uni: UNI, switches: dict) -> Interface:
+        """Get interface from switch by uni"""
+        switch = switches[uni.interface.switch.dpid]
+        interface = switch.interfaces[uni.interface.port_number]
+        return interface
+
+    def handle_topology_update(self, switches: dict):
+        """Handle changes in the topology"""
+        try:
+            interface_a = self.get_interface_from_switch(self.uni_a, switches)
+        except KeyError:
+            id_ = self.uni_a.interface.id
+            log.warning(f"The interface {id_} was not found")
+            return
+
+        try:
+            interface_z = self.get_interface_from_switch(self.uni_z, switches)
+        except KeyError:
+            id_ = self.uni_z.interface.id
+            log.warning(f"The interface {id_} was not found")
+            return
+
+        active, interface = self.is_uni_interface_active(
+            interface_a, interface_z
+        )
+        if self.is_active() != active:
+            if active:
+                self.activate()
+                log.info(f"Activating EVC {self.id}")
```

Comment:
Added both interfaces id in the log. Is this okay? I could not find a way to know which one had been activated recently.

Reply:
@Alopalao, yes, having both there is great considering the event. Follow ups though:

- In the deactivation case, notice that a maintenance can put both UNIs down for instance, so, there's a case where you'd actually have both going on, in the log it would show up one, so maybe to cover all the cases even for down we should also include both the status and status reason of the interfaces in the log?
- Regarding "interfaces are activated" or "interface is inactive", could you adapt it to actually include its `status` (and `status_reason`)? The active state of an interface is just regarding the southbound, so saying that the interface got activated or deactivated won't always be true considering that we also have logical interruptions or admin disable/enable changes. See what I mean? Just minor semantics detail, but let's make it clear for network engineers.

Reply:
Great. Excellent log messages. Let's close this thread.